library/Notifications/Model/Channel.php

```diff
@@ -55,31 +56,43 @@ public function createRelations(Relations $relations)
         $relations->hasMany('contact', Contact::class)
             ->setJoinType('LEFT')
             ->setForeignKey('default_channel_id');
+        $relations->belongsTo('available_channel_type', AvailableChannelType::class)
+            ->setCandidateKey('type');
     }
 
     /**
-     * Fetch and map all the configured channel types to a key => value array
+     * Get the channel icon
+     *
+     * @return Icon
+     */
+    public function getIcon(): Icon
+    {
+        switch ($this->type) {
+            case 'rocketchat':
```

Comment:
Now the channel type can be anything (user defined), and `HasAddress` cannot be changed accordingly.

We had a discussion that the behaviour can be removed, as it is only used to create icons.  They have recommended removing `has_rc` but keeping `has_email` for now as it is the most commonly used.

Since rc is provided here by default, I have added the icon.